### ISSUE TITLE
OIDC Resolving JWK set failed

### ISSUE THREAD

**Issue by Legeril** [DependencyTrack/dependency-track#2696]
### Current Behavior

dual ECS container v4.8.0 deployment
(Thanks for adding proxy support, I at least now can get TO Okta)

On login screen, click the OIDC button. Was redirected to Okta, Authenticate with Okta, Returned to login screen for DependencyTrack.

On further attempts once authenticated through Okta, it just takes straight back to login screen.
In Okta, it shows the token is granted

Logging in as admin  to dtrack shows no user provisioned etc.
DependencyTrack Logs show:
```
ERROR [OidcIdTokenAuthenticator] Resolving JWK Set failed
java.net.ConnectException: Connection timed out
```
I did verify the jwk_uri exists and is accessible ending in /v1/keys

### Steps to Reproduce

Front End has env vars
```
OIDC_ISSUER is set
OIDC_CLIENT_ID is set
OIDC_SCOPE=oepnid
```

API Server has env vars
```
ALPINE_OIDC_ENABLED=true
ALPINE_OIDC_CLIENT_ID is set
ALPINE_OIDC_ISSUER is set
ALPINE_OIDC_USERNAME_CLAIM=username
ALPINE_OIDC_USER_PROVISIONING=true
HTTP_PROXY and HTTPS_PROXY are also set as of v4.8.0 so we at least hit Okta now
```

### Expected Behavior

I would expect to get into the Dashboard.

### Dependency-Track Version

4.8.0

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Database Server Version

15.2

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by nscuro** [comment]
Looks like retrieval of JWKs may still not use the configured proxy.

Please try additionally setting the proxy on the [JVM level](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/net/doc-files/net-properties.html) and see if that works. For the container image, you can use the `JAVA_EXTRA_OPTIONS` environment variable.

https://github.com/DependencyTrack/dependency-track/blob/665c4bcacbb4e5b2de0379c7d37c5adc53cd1e79/src/main/docker/Dockerfile#L20-L21

For example:

```
JAVA_EXTRA_OPTIONS="-Dhttps.proxyHost=proxy.example.com -Dhttps.proxyPort=8080 -Dhttp.noProxyHosts=localhost|host-a.internal|host-b.internal"
```

**Comment by nscuro** [comment]
A fix for the underlying issue has been raised in https://github.com/stevespringett/Alpine/pull/490

This will likely be shipped with a 4.8.1 bugfix release soon.

**Comment by mieliespoor** [comment]
We are looking to upgrade to 4.8, but holding back because of this issue. Does this issue only relate to when the proxy has been set?

**Comment by nscuro** [comment]
@mieliespoor Yes, this issue will only affect you if the communication between the DT API server and the OIDC IdP must go through a proxy.

**Comment by valentijnscholten** [comment]
> I have added the env variable in the container definition
>
> JAVA_EXTRA_OPTIONS="-Dhttps.proxyHost=http://proxy.address -Dhttps.proxyPort=#### -Dhttp.noProxyHosts=list|of|hosts"

It might not help, but the correct name of the env variable is `EXTRA_JAVA_OPTIONS`.

**Comment by Legeril** [comment]
> > I have added the env variable in the container definition
> > JAVA_EXTRA_OPTIONS="-Dhttps.proxyHost=http://proxy.address -Dhttps.proxyPort=#### -Dhttp.noProxyHosts=list|of|hosts"
> 
> It might not help, but the correct name of the env variable is `EXTRA_JAVA_OPTIONS`.

Thank you - I edited the ENV vars to reflect that difference. It got me further than before.

The error is now this but with our actual proxy url.
`Resolving JWK set failed
java.net.UnknownHostException: http://proxy.address`

**Comment by nscuro** [comment]
`proxyHost` must be only the hostname, the `HTTP` protocol is implicit.

So:
```
-Dhttps.proxyHost=proxy.address -Dhttps.proxyPort=1234
```

**Comment by Legeril** [comment]
> `proxyHost` must be only the hostname, the `HTTP` protocol is implicit.
> 
> So:
> 
> ```
> -Dhttps.proxyHost=proxy.address -Dhttps.proxyPort=1234
> ```

@nscuro  thank you. 
The error now seems to be related to OIDC settings we'll need to adjust on our end.
'Unable to assemble complete profile'

I appreciate how much help you've given.

**Comment by nscuro** [comment]
This was fixed in https://github.com/stevespringett/Alpine/pull/490 and will be shipped with 4.8.1.
